packages/stryker/src/process/MutationTestExecutor.ts

```diff
@@ -7,96 +7,73 @@ import { SandboxPool } from '../SandboxPool';
 import TestableMutant from '../TestableMutant';
 import TranspiledMutant from '../TranspiledMutant';
 import StrictReporter from '../reporters/StrictReporter';
-import MutantTranspiler from '../transpiler/MutantTranspiler';
+import { MutantTranspileScheduler } from '../transpiler/MutantTranspileScheduler';
 import { Logger } from 'stryker-api/logging';
 import { tokens, commonTokens } from 'stryker-api/plugin';
 import { coreTokens } from '../di';
-import InputFileCollection from '../input/InputFileCollection';
 
 export class MutationTestExecutor {
   public static inject = tokens(
     commonTokens.logger,
-    coreTokens.inputFiles,
     coreTokens.reporter,
-    coreTokens.mutantTranspiler,
+    coreTokens.mutantTranspileScheduler,
     coreTokens.sandboxPool);
   constructor(
     private readonly log: Logger,
-    private readonly input: InputFileCollection,
     private readonly reporter: StrictReporter,
-    private readonly mutantTranspiler: MutantTranspiler,
+    private readonly mutantTranspileScheduler: MutantTranspileScheduler,
     private readonly sandboxPool: SandboxPool) {
   }
 
   public async run(allMutants: ReadonlyArray<TestableMutant>): Promise<MutantResult[]> {
-    const transpiledFiles = await this.mutantTranspiler.initialize(this.input.files);
-    const result = await this.runInsideSandboxes(
-      this.sandboxPool.streamSandboxes(transpiledFiles),
-      this.mutantTranspiler.transpileMutants(allMutants));
+    const results = await this.mutantTranspileScheduler.scheduleTranspileMutants(allMutants).pipe(
+      // Test the mutant (either early result, or actually test it in a sandbox)
+      flatMap(this.mutationTest),
+      tap(this.reportResult),
+      // Signal the mutant transpiler that there is another slot open for transpiling
+      tap(this.mutantTranspileScheduler.scheduleNext),
+      toArray(),
+      tap(this.reportAll)
+    ).toPromise();
+
+    // TODO: Let typed inject dispose of sandbox pool
```

Comment:
Yes it is. Once all disposable instances are created with `typed-inject` we can use it to dispose all our stuff. As for now, we cannot do that, so that's what this hack is for.

NOTE: this functionality didn't change. Disposing was always done at this point.